controllers/teleport_master_controller/teleport_master_controller.cpp

```diff
@@ -20,14 +20,34 @@
 // Special thanks to Hamburg Bit-Bots for ideas for this controller
 // https://github.com/bit-bots/wolfgang_robot/blob/feature/recognition/wolfgang_webots_sim/src/wolfgang_webots_sim/webots_camera_controller.py
 
+#include <Eigen/Core>
+#include <Eigen/Geometry>
 #include <array>
 #include <cmath>
+#include <filesystem>
 #include <fstream>
+#include <iomanip>
 #include <iostream>
 #include <random>
+#include <sstream>
+#include <thread>
+#include <webots/Camera.hpp>
+#include <webots/Motor.hpp>
+#include <webots/PositionSensor.hpp>
+#include <webots/Robot.hpp>
 #include <webots/Supervisor.hpp>
 #include <yaml-cpp/yaml.h>
 
+
+std::string padLeft(int number, int width) {
+    std::stringstream ss;
+    ss << std::setw(width) << std::setfill('0') << number;
```

Comment:
I think this is actually padding it with ascii 0, instead of null, which I am pretty sure should be `'\0'`.

Reply:
This isn't padding empty characters, it's meant to pad with 0s. It's for the naming of the data files.